packages/dev/core/src/Misc/logger.ts

```diff
@@ -55,23 +63,14 @@ export class Logger {
         return entry.current <= entry.limit;
     }
 
-    private static _GenerateLimitMessage(message: string, messageType: number): void {
+    private static _GenerateLimitMessage(message: string, level: number = 1): void {
         const entry = Logger._LogLimitOutputs[message];
         if (!entry || !Logger.MessageLimitReached) {
             return;
         }
+        const type = this._Levels[level];
         if (entry.current === entry.limit) {
-            switch (messageType) {
-                case 0:
-                    Logger.Log(Logger.MessageLimitReached.replace(/%LIMIT%/g, "" + entry.limit).replace(/%TYPE%/g, "log"));
-                    break;
-                case 1:
-                    Logger.Warn(Logger.MessageLimitReached.replace(/%LIMIT%/g, "" + entry.limit).replace(/%TYPE%/g, "warning"));
-                    break;
-                case 2:
-                    Logger.Error(Logger.MessageLimitReached.replace(/%LIMIT%/g, "" + entry.limit).replace(/%TYPE%/g, "error"));
-                    break;
-            }
+            Logger.Log(Logger.MessageLimitReached.replace(/%LIMIT%/g, "" + entry.limit).replace(/%TYPE%/g, type.name ?? ""));

```

Comment:
Shouldn't you use `Logger.Log` / `Logger.Warn` / `Logger.Error` depending on the level?

Reply:
i'll fix that, great catch!